src/scope/scope.js

```diff
@@ -538,6 +538,91 @@ export default class Scope {
     return Promise.all(versionDependenciesArr.map(versionDependencies => versionDependencies.toConsumer(this.objects)));
   }
 
+  /**
+   * Remove or deprecate single component
+   * @removeComponent - boolean - true if you want to remove component
+   */
+  async removeSingle(bitId: BitId, removeComponent: boolean): Promise<string> {
+    if (removeComponent) {
+      await this.sources.clean(bitId);
+      return bitId.toStringWithoutVersion();
+    }
+    const component = await this.sources.get(bitId);
+    component.deprecated = true;
+    this.objects.add(component);
+    await this.objects.persist();
+    return component.id();
+  }
+
+  /**
+   * findDependentBits
+   * foreach component in array find the componnet that uses that component
+   */
+  async findDependentBits(bitIds: Array<BitId>): Promise<Array<object>> {
+    const stagedComponents = await this.listStage();
+    return new Promise((resolve) => {
+      const dependentBits = {};
+      bitIds.forEach((bitId) => {
```

Comment:
You can write this logic in a much better way using lodash/ramda but it's not that important.